src/components/KYCInfoFieldSection.tsx

```diff
@@ -51,13 +51,23 @@ function KYCInfoFieldSection({
     const requiredFields = Object.entries(kycSchemaMetadata)
       .filter(([_, metadata]) => metadata.required)
       .map(([field, _]) => field)
-    const fieldValues = requiredFields.map((field) => kycInfo[field])
-    if (!fieldValues.every((value) => !!value)) {
-      setErrorMessage(undefined)
-      setSubmitDisabled(true)
-      return
+    for (const requiredField of requiredFields) {
+      const fieldMetadata = kycSchemaMetadata[requiredField]
+      if (fieldMetadata.group) {
+        const groupValue = kycInfo[fieldMetadata.group]
+        if (typeof groupValue === 'object' && !groupValue[requiredField]) {
```

Comment:
seems like if the groupValue was something other than an object, something would be very wrong, and we would fail silently here. might be worth adding some safeguard for that-- could just be a unit test on our kyc schemas

Reply:
This is actually somewhat tricky to test for - just testing our KYC schema won't suffice. This `if` statement is checking if the constructed `kycInfo` object has a field that is a string, but _ought_ to be an object based on the KYC schema metadata. The `kycInfo` info object's shape directly stems from the KYC metadata, and is constructed dynamically based off the logic in the `setKycInfoWrapper` helper. Checking the schemas within the `componenets/kycSchemas/` directory won't suffice to test this. The `typeof group === 'string'` check in `setKycInfoWrapper` is actually a safeguard that would prevent this case from occuring in the first place.